packages/react/src/components/NavBar/navBar.test.tsx

```diff
@@ -0,0 +1,166 @@
+      <NavItem active={false}>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'false');
+  expect(screen.getByRole('button')).toHaveClass('NavBar__trigger');
+  expect(screen.getByRole('button')).not.toHaveClass('NavBar__trigger--active');
+  expect(screen.queryByRole('listitem')).not.toBeInTheDocument();
+  expect(screen.getByRole('navigation').querySelector('.Icon')).toHaveClass(
+    'Icon',
+    'Icon--hamburger-menu'
+  );
+});
+
+test('should render navTriggerLabel properly', () => {
+  render(
+    <NavBar collapsed navTriggerLabel="I am a label">
+      <NavItem active={false}>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  expect(screen.getByRole('button')).toHaveAccessibleName('I am a label');
+});
+
+test('should render aria-controls prop on trigger button', () => {
+  render(
+    <NavBar collapsed propId="someid">
+      <NavItem active={false}>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  expect(screen.getByRole('button')).toHaveAttribute('aria-controls', 'someid');
+});
+
+test('should render aria-controls prop on trigger button', () => {
+  render(
+    <NavBar collapsed>
+      <NavItem active>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  fireEvent.click(screen.getByRole('button'));
+  expect(screen.getByText('first item')).toBeInTheDocument();
+});
+
+test('should hide NavItems after pressing escape key', () => {
+  render(
+    <NavBar collapsed>
+      <NavItem active>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  fireEvent.click(screen.getByRole('button'));
+  fireEvent.keyDown(screen.getByRole('list'), { key: 'Escape' });
+  expect(screen.queryByRole('listitem')).not.toBeInTheDocument();
+});
+
+test('should not hide NavItems after pressing other keys', () => {
+  render(
+    <NavBar collapsed>
+      <NavItem active>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  fireEvent.click(screen.getByRole('button'));
+  fireEvent.keyDown(screen.getByRole('list'), { key: 'Home' });
+  expect(screen.queryByRole('listitem')).toBeInTheDocument();
+});
+
+test('should hide NavItems when focusing outside nav', () => {
+  render(
+    <NavBar collapsed>
+      <NavItem active>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  fireEvent.click(screen.getByTestId('link'));
+  fireEvent.focusIn(screen.getByTestId('link'));
+  expect(screen.queryByRole('listitem')).not.toBeInTheDocument();
+});
+
+test('should not hide NavItems when focusing inside nav', () => {
+  render(
+    <NavBar collapsed>
+      <NavItem active>
+        <p>first item</p>
+      </NavItem>
+    </NavBar>
+  );
+
+  fireEvent.click(screen.getByRole('button'));
+  fireEvent.focusIn(screen.getByRole('listitem'));
+  expect(screen.getByRole('listitem')).toBeInTheDocument();
+});
+
+test('returns no axe violations', async () => {
```

Comment:
Only thing missing is running axe against both collapsed/not-collapsed states.

Reply:
No worries! I understand you're very busy. Anything for you and Deque. 😄